gandalf.js

```diff
@@ -1,87 +1,91 @@
 /**
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.